From b9f512c2ae381eeba9a4a56490fba352e2041517 Mon Sep 17 00:00:00 2001
From: Elias Kuiter <info@elias-kuiter.de>
Date: Thu, 21 Aug 2025 13:18:26 +0200
Subject: [PATCH 1/2] Rename reset command to uninstall with confirmation
 prompt

- Rename command-reset to command-uninstall in docker.sh and main.sh
- Add confirmation prompt before removing Docker containers and images
- Update documentation to reflect new command name
- Update KClause to newer commit bd3e732
---
 src/docker/kclause/Dockerfile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/docker/kclause/Dockerfile b/src/docker/kclause/Dockerfile
index 0736872..2beb6cd 100644
--- a/src/docker/kclause/Dockerfile
+++ b/src/docker/kclause/Dockerfile
@@ -19,7 +19,7 @@ WORKDIR /home
 # we fix z3 to an older version although we do not need it here, as 4.12.2.0 does not compile correctly
 RUN git clone https://github.com/paulgazz/kmax.git \
   && cd kmax \
-  && git checkout d0fe92db13ca97140385cd39977d937d1ee92d4c \
+  && git checkout bd3e73239c83ac04e4ea4e3ce12c872475490658 \
   && sed -i 's/z3-solver/z3-solver==4.11.2/' setup.py \
   && python3 setup.py install
 
-- 
2.49.1


From 4c21da38dd96e9d3d27d5cfbe2456e913bbdf0cc Mon Sep 17 00:00:00 2001
From: Elias Kuiter <info@elias-kuiter.de>
Date: Thu, 21 Aug 2025 14:31:50 +0200
Subject: [PATCH 2/2] updated KMax

---
 src/docker/kclause/Dockerfile | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/src/docker/kclause/Dockerfile b/src/docker/kclause/Dockerfile
index 2beb6cd..1480432 100644
--- a/src/docker/kclause/Dockerfile
+++ b/src/docker/kclause/Dockerfile
@@ -1,12 +1,15 @@
-FROM ubuntu:18.04
+FROM ubuntu:22.04
 
-# install (tested) dependencies of kmax
+# kmax and system (e.g., Linux) dependencies
+ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get update && apt-get install -y --allow-unauthenticated \
   git \
   python3-setuptools \
+  pipx \
   python3-dev \
   flex \
   bison \
+  bc \
   libssl-dev \
   libelf-dev \
   wget \
@@ -23,10 +26,6 @@ RUN git clone https://github.com/paulgazz/kmax.git \
   && sed -i 's/z3-solver/z3-solver==4.11.2/' setup.py \
   && python3 setup.py install
 
-# install dependencies of analyzed projects
-RUN apt-get update && apt-get install -y --allow-unauthenticated \
-  build-essential
-
 # copy and set up
 # - kextractor (which is compiled against the projects' Kconfig implementations)
 # - IO scripts
-- 
2.49.1

